### PR TITLE
rdar://107033737 (Expose `PackageVersionMetadata` from `PackageSearchClient`)

### PR DESCRIPTION
**Pull request by neonichu**
This exposes all of the version-specific metadata, not just the readme URL. For convenience, this is still flattened and only the metadata for the latest version is available, but I have also added a `latestVersion` property if we are taking metadata from a specific version to indicate where it is from.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/PackageMetadata/PackageMetadata.swift [hunk 1]
```diff
@@ -30,19 +30,60 @@ public struct Package {
...
+        public let name: String
+        public let type: String
+        public let checksum: String?
+        public let signing: Signing?
+    }
+
+    public struct Signing: Sendable {
+        public let signatureBase64Encoded: String
+        public let signatureFormat: String
+    }
```
Comment (tomerd) [inline Sources/PackageMetadata/PackageMetadata.swift:43]: should we translate this into the SigningEntity? or is the client expect ted to do that?

Comment (neonichu) [reply Sources/PackageMetadata/PackageMetadata.swift:43]: IDK, this just matches what is available today in `PackageVersionMetadata` when talking to the registry client.

Comment (tomerd) [reply Sources/PackageMetadata/PackageMetadata.swift:43]: maybe a good enough start

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

**Review by tomerd — APPROVED** [review]

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

**Pull request merged**